src/platform_impl/linux/taskbar/unity.rs

```diff
@@ -0,0 +1,131 @@
+use zbus::{blocking::Connection, MessageBuilder};
+
+pub struct Manager {
+  conn: Connection,
+  app_uri: String,
+  count: i32,
+  progress: f64,
+  progress_visible: bool,
+  count_visible: bool,
+  urgent: bool,
+  dirty_count: bool,
+  dirty_progress: bool,
+  dirty_progress_visible: bool,
+  dirty_count_visible: bool,
+  dirty_urgent: bool,
+}
+
+enum DataTypes<'a> {
+  Str(&'a str),
+  String(&'a String),
+  Bool(&'a bool),
+  Number(&'a i32),
+  Float(&'a f64),
+}
+
+impl Manager {
+  pub fn new(app_uri: String) -> Result<Self, zbus::Error> {
+    let conn = Connection::session()?;
+    let mut m = Self {
+      conn,
+      app_uri,
+      progress: 0.0,
+      count: 0,
+      progress_visible: false,
+      count_visible: false,
+      urgent: false,
+      dirty_count: true,
+      dirty_progress: true,
+      dirty_progress_visible: true,
+      dirty_count_visible: true,
+      dirty_urgent: true,
+    };
+
+    m.update().unwrap_or(());
+    Ok(m)
+  }
+
+  fn update(&mut self) -> Result<(), Box<dyn std::error::Error>> {
+    let mut properties: Vec<DataTypes> = Vec::new();
+
+    properties.push(DataTypes::String(&self.app_uri));
+
+    if self.dirty_progress {
+      self.dirty_progress = false;
+
+      properties.push(DataTypes::Str("progress"));
+
+      properties.push(DataTypes::Float(&self.progress));
+    }
+    if self.dirty_count {
+      self.dirty_count = false;
+
+      properties.push(DataTypes::Str("count"));
+      properties.push(DataTypes::Number(&self.count));
+    }
+    if self.dirty_progress_visible {
+      self.dirty_progress_visible = false;
+
+      properties.push(DataTypes::Str("progress-visible"));
+      properties.push(DataTypes::Bool(&self.progress_visible));
+    }
+    if self.dirty_count_visible {
+      self.dirty_count_visible = false;
+
+      properties.push(DataTypes::Str("count-visible"));
+      properties.push(DataTypes::Bool(&self.count_visible));
+    }
+    if self.dirty_urgent {
+      self.dirty_urgent = false;
+
+      properties.push(DataTypes::Str("urgent"));
+      properties.push(DataTypes::Bool(&self.urgent));
+    }
+
+    if !properties.is_empty() {
+      let mapped = properties
```

Comment:
I guess `mapped` is the reason why this zbus implementation is not working.

The type of `mapped`'s properties is mapping to `string` only, so the server side may not find the method to call since the signature is different.

Maybe you could try using `zvarient` provided by zbus and let them do the Type transformation.
The `Dictionary` strcut in this [FAQ](https://dbus2.github.io/zbus/faq.html) seems fit the need of `properties`

Reply:
Something like this

```rust
#[derive(Default, SerializeDict, DeserializeDict, Type, PartialEq, Debug)]
#[zvariant(signature = "dict")]
struct UpdateProperties {
  progress: Option<f64>,
  #[zvariant(rename = "progress-visible")]
  progress_visible: Option<bool>,
  count: Option<i32>,
  #[zvariant(rename = "count-visible")]
  count_visible: Option<bool>,
  urgent: Option<bool>
}

let mut properties = UpdateProperties::default();

properties.progress = Some(self.progress);

let body = (self.app_uri.to_owned(), properties);
let signal =
      MessageBuilder::signal("/", "com.canonical.Unity.LauncherEntry", "Update")?.build(&body)?;
```

Reply:
I also found another varient using HashMap
I am kinda researching.

Will work on an impl today

Reply:
Main Problem I am facing currently is that my area lies in the storm region of India. And Electricity is behaving like anything.. (and I own a desktop)